kolibri/plugins/learn/assets/src/vue/assessment-wrapper/index.vue

```diff
@@ -0,0 +1,441 @@
+          if (attempt.hinted) {
+            return 'hint';
+          }
+          return attempt.correct ? 'right' : 'wrong';
+        }).reverse();
+      },
+      mOfNMasteryModel() {
+        return MasteryModelGenerators[this.masteryModel.type](
+          this.assessmentIds, this.masteryModel);
+      },
+      totalCorrectRequiredM() {
+        return this.mOfNMasteryModel.m;
+      },
+      attemptsWindowN() {
+        return this.mOfNMasteryModel.n;
+      },
+      exerciseProgress() {
+        if (this.pastattempts) {
+          if (this.pastattempts.length > this.attemptsWindowN) {
+            return Math.min(
+              this.pastattempts.slice(
+                0,
+                this.attemptsWindowN
+              ).reduce((a, b) => a + b.correct, 0) / this.totalCorrectRequiredM,
+              1.0
+              );
+          }
+          return Math.min(
+            this.pastattempts.reduce((a, b) => a + b.correct, 0) / this.totalCorrectRequiredM,
+            1.0
+            );
+        }
+        return 0.0;
+      },
+      success() {
+        return this.exerciseProgress === 1.0;
+      },
+    },
+    vuex: {
+      actions: {
+        initMasteryLogAction: actions.initMasteryLog,
+        createDummyMasteryLogAction: actions.createDummyMasteryLog,
+        saveMasteryLogAction: actions.saveMasteryLog,
+        setMasteryLogCompleteAction: actions.setMasteryLogComplete,
+        createAttemptLogAction: actions.createAttemptLog,
+        saveAttemptLogAction: actions.saveAttemptLog,
+        updateMasteryAttemptStateAction: actions.updateMasteryAttemptState,
+        updateAttemptLogInteractionHistoryAction: actions.updateAttemptLogInteractionHistory,
+        updateExerciseProgress: actions.updateExerciseProgress,
+      },
+      getters: {
+        userkind: (state) => state.core.session.kind,
+        totalattempts: (state) => state.core.logging.mastery.totalattempts,
+        pastattempts: (state) => state.core.logging.mastery.pastattempts,
+        userid: (state) => state.core.session.user_id,
+        content: (state) => state.pageState.content,
+        assessmentIds: (state) => state.pageState.content.assessmentIds,
+        masteryModel: (state) => state.pageState.content.masteryModel,
+        randomize: (state) => state.pageState.content.randomize,
+      },
+    },
+  };
+
+</script>
+
+
+<style lang="stylus" scoped>
+
+  @require '~kolibri.styles.definitions'
+
+  .message
+    color: grey
+    padding: 16px
+    font-size: 14px
+    @media screen and (max-width: $portrait-breakpoint)
+      position: relative
+      text-align: center
+      clear: both
+      top: 40px
+      font-size: 12px
+      margin-top: 0
+      padding: 0
+
+  .attemptprogress
+    position: absolute
+    padding-left: 14px
+    top: 38px
+    @media screen and (max-width: $portrait-breakpoint)
+      top: 0
+      padding-left: 0
+      left: 50%
+      transform: translate(-50%, 0)
+
+  #attemptprogress-container
```

Comment:
note that much of this code existed before this PR in a different file. We may want to merge in and create issues for this items